geonode/upload/uploadhandler.py

```diff
@@ -0,0 +1,215 @@
+
+                # This is a post field, we can just set it in the post
+                if transfer_encoding == "base64":
+                    raw_data = field_stream.read(size=read_size)
+                    num_bytes_read += len(raw_data)
+                    try:
+                        data = base64.b64decode(raw_data)
+                    except binascii.Error:
+                        data = raw_data
+                else:
+                    data = field_stream.read(size=read_size)
+                    num_bytes_read += len(data)
+
+                # Add two here to make the check consistent with the
+                # x-www-form-urlencoded check that includes '&='.
+                num_bytes_read += len(field_name) + 2
+                if (
+                    settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None
+                    and num_bytes_read > settings.DATA_UPLOAD_MAX_MEMORY_SIZE
+                ):
+                    raise RequestDataTooBig("Request body exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE.")
+
+                _post.appendlist(field_name, force_str(data, encoding, errors="replace"))
+            elif item_type == FILE:
+                # NOTE: Parse files WITHOUT a stream.
+                # This is a file, use the handler...
+                file_name = disposition.get("filename")
+                if file_name:
+                    file_name = force_str(file_name, encoding, errors="replace")
+                    file_name = self.sanitize_file_name(file_name)
+                if not file_name:
+                    continue
+
+                content_type, content_type_extra = meta_data.get("content-type", ("", {}))
+                content_type = content_type.strip()
+                charset = content_type_extra.get("charset")
+                content_length = None
+
+                self.new_file(field_name, file_name, content_type, content_length, charset, content_type_extra)
+
+                # Handle file upload completions on next iteration.
+                old_field_name = field_name
+            else:
+                # If this is neither a FIELD or a FILE, just exhaust the stream.
+                exhaust(stream)
+
+        # Make sure that the request data is all fed
+        exhaust(input_data)
+
+        _post._mutable = False
+        return _post, _files
+
+    def sanitize_file_name(self, file_name):
+        """
+        Same as django.http.multipartparser.MultiPartParser.sanitize_file_name
+        so it's not needed to instantiate a MultiPartParser object just to
+        sanitize the filename of an upload.
+        """
+        file_name = html.unescape(file_name)
+        file_name = file_name.rsplit("/")[-1]
+        file_name = file_name.rsplit("\\")[-1]
+
+        if file_name in {"", ".", ".."}:
+            return None
+        return file_name
+
+    def _get_max_size(self):
+        max_size_db_obj, created = UploadSizeLimit.objects.get_or_create(
```

Comment:
From my point of view, will be better to create by default the UploadSizeLimit with the default value during the migration phase.
Also here we are hiding a possible object creation inside a get method which could not be a good pattern.
A possible solution could be using the exists method of the queryset and in the case of Not exists object:

1. raise an exception
2. call creation method (maybe a method of the model like UploadSizeLimit.objects.create_default_limit())

Reply:
Updated ✅ link: https://github.com/GeoNode/geonode/pull/8512/files#diff-7d1f963cfdf3df10fcd4c9edd023d0e095745fff6477219eba74ff40889185b2R173-R178